### PR TITLE
allow ImportError exception to avoid missing boto3 during __init__

### PR DESCRIPTION
**Pull request by syahdeini**
I received this error 
image
The problem is when boto3 is not installed, then it will throw null AttributeError. 
I believe we should allow the ImportError and the try-catch expectation is not really useful therefore I delete it.

### CODE REVIEW

File: great_expectations/datasource/generator/s3_generator.py [hunk 1]
```diff
@@ -94,10 +91,7 @@ def __init__(self,
             boto3_options = {}
         self._max_keys = max_keys
         self._iterators = {}
-        try:
-            self._s3 = boto3.client('s3', **boto3_options)
-        except TypeError:
```
Comment (syahdeini) [inline great_expectations/datasource/generator/s3_generator.py:99]: It is also possible to use AttributeError here, but this module will need to use boto3 nevertheless so why not make it fail on the first place.

**Comment by Aylr** [comment]
@syahdeini Thanks for looking at this. In an effort to allow people to run great expectations with as few dependencies as possible, we want to allow this module to at least import without requiring boto3. However you make an excellent point that the `TypeError` should be changed to an `AttributeError`. Would you like to make this change (and leave the try catch around the `import boto3`?

**Review by Aylr — CHANGES_REQUESTED** [review]
@syahdeini Thanks for looking at this. In an effort to allow people to run great expectations with as few dependencies as possible, we want to allow this module to at least import without requiring boto3. However you make an excellent point that the `TypeError` should be changed to an `AttributeError`. Would you like to make this change (and leave the try catch around the `import boto3`?

**Comment by syahdeini** [comment]
@Aylr Done, Thanks for the opportunity and the great works!

**Review by Aylr — APPROVED** [review]
Hey, thanks again this was a great catch!

**Comment by Aylr** [comment]
@kyleaton Can you check out the cla bot issues on this PR?

**Comment by Aylr** [comment]
@syahdeini Will you please sign our Contributor License Agreement so we can merge this? https://docs.greatexpectations.io/en/latest/contributing/miscellaneous.html#contributor-license-agreement-cla

**Comment by kyleaton** [comment]
@cla-bot check

**Comment by kyleaton** [comment]
@cla-bot check

**Comment by kyleaton** [comment]
@cla-bot check

**Pull request merged**